src/services/Notion.ts

```diff
@@ -0,0 +1,67 @@
+import { Client } from '@notionhq/client';
+import { InterfaceAPI } from './types';
+import { Config } from '../types';
+
+/** Class representing an interaction with Notion. */
+export class Notion implements InterfaceAPI {
+  public static instance: Notion;
+  public notion: Client | null = null;
+  public configuration?: Config;
+  /**
+   * Create and returns a Notion instance.
+   *
+   * @returns {Notion}
+   */
+  constructor() {
+    if (!Notion.instance) {
+      Notion.instance = this;
+    }
+
+    return Notion.instance;
```

Comment:
I'm not sure about new Notion(configuration) with this:
```
export default new Notion();
// or
export const notion = new Notion();
```
And if we use instance inside it it will be singleton always. In this variation Notion became singleton outside class. 
In classical way we have getInstance and have smth like `Notion.getInstance().do()` but `static instance` just hide this `getInstance()`
But if it need i will change it 🙂

Reply:
I see no reason to create a singletone. In a good module design only a specific part of the system should have an access to that service. Not all of them.

Reply:
I see no reason to create a singletone. In a good module design only a specific part of the system should have an access to that service. Not all of them.